keel-tagging-plugin/src/main/kotlin/com/netflix/spinnaker/keel/tagging/ResourceTagger.kt

```diff
@@ -93,10 +93,12 @@ class ResourceTagger(
   fun onDeleteEvent(event: DeleteEvent) {
     if (event.resourceName.shouldTag()) {
       log.debug("Persisting no tag desired for resource {} because it is no longer managed", event.resourceName.toString())
+      val entityRef = event.resourceName.toEntityRef()
       val spec = KeelTagSpec(
         keelId = event.resourceName.toString(),
-        entityRef = event.resourceName.toEntityRef(),
-        tagState = TagNotDesired(startTime = clock.millis())
+        entityRef = entityRef,
+        tagState = TagNotDesired(startTime = clock.millis()),
+        application = entityRef.application
```

Comment:
specifically to comment above on KeelTagSpec, this seems redundant to supply here